core/pool.go

```diff
@@ -9,7 +9,7 @@ import (
 type Pool struct {
 }
 
-func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig) {
+func (p Pool) Start(engines []Engine, provider RequestProvider, metricsManager MetricsManager, config ChukonuConfig, fuse chan bool, ack chan bool) {
```

Comment:
Agree. One way to solve this I had in mind is to have multiple checkpointing of resource usage and gradually increase goroutine spawned, since we want to utilize available resource in a container as much as we can but not overwhelm the container with too many goroutine at once.